src/rust-crypto/rust-crypto.ts

```diff
@@ -247,15 +247,33 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
     /**
      * Implementation of {@link CryptoApi.userHasCrossSigningKeys}.
      */
-    public async userHasCrossSigningKeys(): Promise<boolean> {
-        const userId = new RustSdkCryptoJs.UserId(this.userId);
-        /* make sure we have an *up-to-date* idea of the user's cross-signing keys. This is important, because if we
-         * return "false" here, we will end up generating new cross-signing keys and replacing the existing ones.
-         */
-        const request = this.olmMachine.queryKeysForUsers([userId]);
-        await this.outgoingRequestProcessor.makeOutgoingRequest(request);
-        const userIdentity = await this.olmMachine.getIdentity(userId);
-        return userIdentity !== undefined;
+    public async userHasCrossSigningKeys(userId = this.userId, downloadUncached = false): Promise<boolean> {
+        const rustTrackedUsers: Set<RustSdkCryptoJs.UserId> = await this.olmMachine.trackedUsers();
+        const rustTrackedUser = Array.from(rustTrackedUsers).find((rustUserId) => userId === rustUserId.toString());
+
+        if (rustTrackedUser) {
+            /* make sure we have an *up-to-date* idea of the user's cross-signing keys. This is important, because if we
+             * return "false" here, we will end up generating new cross-signing keys and replacing the existing ones.
+             */
```

Comment:
I think this is only important for our own user (ie, if `userId === this.userId`). At least, the comment doesn't make any sense for other users.